Lib/random.py

```diff
@@ -722,20 +667,98 @@ def weibullvariate(self, alpha, beta):
         u = 1.0 - self.random()
         return alpha * (-_log(u)) ** (1.0/beta)
 
+
+## --------------- Mersene Twister Source ---------------------------
+
+class Random(_random.Random, BaseRandom):
+    """Mersenne Twister Random number generator used by bound module functions.
+
+    Used to instantiate instances of Random to get generators that don't
+    share state.
+    """
+
```

Comment:
Substantially, all of the logic  below this line should be in the parent class.  It is not MT specific.  Also, existing subclasses may reasonably be relying on this logic being present.   The can override the methods if they want, but they can also inherit or extend as needed.

Reply:
Not all RNG have a state. Example: SystemRandom.

Reply:
If a project currently inherits from random.Random, it can continue to inherit from random.Random (rather than inherit from random.BaseRandom) to still inherit these methods. My PR doesn't prevent that.

getstate() and setstate() are very specific to Mersenne Twister: they call _random.Random.getstate() and _random.Random.setstate() which serialize/deserialize the Mersenne Twister internal state.

I'm tempted to keep seed() features in BaseRandom (accept types other than int), but there is a practical issue: BaseRandom.seed() cannot call super().seed(), since it *is* the base class (it has no parent). I'm not comfortable to add another method like _seed() which would be called by seed().

The seed() version is tightly coupled to random.Random history and to Python implementation of Mersenne Twister.

Also seed() emits a deprecation warning. I would prefer to not write a new class which emits a DeprecationWarning since day 1, but start from scratch with a clean API.